### PR TITLE
.NET core 3.1: Refactors all projects to use latest LTS version

### LINKED ISSUE #2150: .NET 3.1 upgrades
All the tests, benchmarks, and sample projects should be updated to use the latest [.NET 5 version](https://dotnet.microsoft.com/platform/support/policy/dotnet-core).

### PR DESCRIPTION
**Pull request by j82w**
# Pull Request Template

## Description

.NET core 2.1 support ends this month. This is bumping all project to 3.1 which is the current LTS version.

https://dotnet.microsoft.com/platform/support/policy/dotnet-core

## Type of change

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #2150

### CODE REVIEW

File: Microsoft.Azure.Cosmos.Samples/Usage/ContainerManagement/ContainerManagement.csproj [hunk 1]
```diff
@@ -1,8 +1,8 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp2.1</TargetFramework>
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos.Samples/Usage/ContainerManagement/ContainerManagement.csproj:5]: nit: Shouldde have this value in the props file to make future bumps smaller?

**Review by ealsur — DISMISSED** [review]

File: Microsoft.Azure.Cosmos.Samples/Usage/ContainerManagement/ContainerManagement.csproj (hunk 1 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos.Samples/Usage/ContainerManagement/ContainerManagement.csproj:5]: The samples are meant to be simple and work on their own. If we put it in the props files it requires them to understand the logic.

**Review by philipthomas-microsoft — APPROVED** [review]

File: Microsoft.Azure.Cosmos/src/Resource/Settings/IncludedPath.cs [hunk 2]
```diff
@@ -33,7 +33,7 @@ public sealed class IncludedPath
         /// The collection of the <see cref="Index"/> objects to be applied for this included path.
         /// </value>
         [JsonProperty(PropertyName = Constants.Properties.Indexes)]
-        internal Collection<Index> Indexes { get; set; } = new Collection<Index>();
+        internal Collection<Cosmos.Index> Indexes { get; set; } = new Collection<Cosmos.Index>();
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/IncludedPath.cs:36]: How did it impact E2E? 
Are there any UT gaps?

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/BaselineTest/TestBaseline/PartitionKeyHashBaselineTest.Numbers.xml [hunk 3]
```diff
@@ -12,7 +12,7 @@
   <Result>
     <Input>
       <Description>negative zero</Description>
-      <PartitionKeyValue>0</PartitionKeyValue>
+      <PartitionKeyValue>-0</PartitionKeyValue>
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/BaselineTest/TestBaseline/PartitionKeyHashBaselineTest.Numbers.xml:15]: These are scary. Why '-' at front?

**Review by kirankumarkolli — DISMISSED** [review]

File: Microsoft.Azure.Cosmos/src/Resource/Settings/IncludedPath.cs (hunk 2 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/IncludedPath.cs:36]: .NET 3.1 introduces it's own Index type causing a namespace type conflict. I added the namespace to remove the ambiguity.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/BaselineTest/TestBaseline/PartitionKeyHashBaselineTest.Numbers.xml (hunk 3 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/BaselineTest/TestBaseline/PartitionKeyHashBaselineTest.Numbers.xml:15]: This a breaking change with .NET core 3.0. https://devblogs.microsoft.com/dotnet/floating-point-parsing-and-formatting-improvements-in-net-core-3-0/

**Review by kirankumarkolli — APPROVED** [review]

File: Microsoft.Azure.Cosmos/src/Resource/Settings/IncludedPath.cs (hunk 2 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/IncludedPath.cs:36]: .NET core 3.1 introduces a new Index type that causes type conflicts. There are no gaps in test.

**Pull request merged**